### PR TITLE
Add fnmatch to libsysbase.

### LINKED ISSUE #11: devkitA64: Build posix into libc (for fnmatch)
I have at least two libraries I'm trying to build that are trying to use fnmatch. One is configure based and the other is CMake based. The CMake based one is assuming the feature is there and not checking it (which sucks). For both it finds fnmatch.h but gets a linker error later. I can build it into my library (which is a bad idea) or hack the configure script output to not define HAVE_FNMATCH_H like so:
```
sed -i -e 's/#define HAVE_FNMATCH_H 1//' aarch64-none-elf/config.h
```

That is not ideal though. Is there a downside to building the posix implementation into libc? I see that opendir is still in libsysbase (hence the HAVE_OPENDIR define) after I make this change and half the posix files can be disabled or don't build anyway.

```diff
diff --git a/newlib/configure.host b/newlib/configure.host
index d536e0d..5c971a1 100644
--- a/newlib/configure.host
+++ b/newlib/configure.host
@@ -409,6 +409,9 @@ case "${host}" in
        sys_dir=a29khif
        signal_dir=
        ;;
+  aarch64*-*-*)
+    posix_dir=posix
+    ;;
   arm*-*-*)
        sys_dir=arm
        if [ "x${newlib_may_supply_syscalls}" = "xno" ] ; then
@@ -627,7 +630,7 @@ newlib_cflags="${newlib_cflags} -DCLOCK_PROVIDED -DMALLOC_PROVIDED -DEXIT_PROVID
        default_newlib_io_c99_formats="yes"
        default_newlib_io_long_long="yes"
        default_newlib_io_pos_args="yes"
-       newlib_cflags="${newlib_cflags} -D__DYNAMIC_REENT__ -DREENTRANT_SYSCALLS_PROVIDED -D__DEFAULT_UTF8__"
+       newlib_cflags="${newlib_cflags} -DHAVE_OPENDIR -D__DYNAMIC_REENT__ -DREENTRANT_SYSCALLS_PROVIDED -D__DEFAULT_UTF8__"
        newlib_cflags="${newlib_cflags} -ffunction-sections -fdata-sections"
        syscall_dir=syscalls
        ;;

```

### PR DESCRIPTION
**Pull request by compomega**
Closes #11

### CODE REVIEW

File: libgloss/libsysbase/fnmatch.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+#if defined(LIBC_SCCS) && !defined(lint)
+static char sccsid[] = "@(#)fnmatch.c	8.2 (Berkeley) 4/16/94";
+#endif /* LIBC_SCCS and not lint */
+
+/*
+ * Function fnmatch() as specified in POSIX 1003.2-1992, section B.6.
+ * Compares a filename or pathname to a pattern.
+ */
+
```
Comment (WinterMute) [inline libgloss/libsysbase/fnmatch.c]: Add #define _GNU_SOURCE here

Comment (compomega) [reply libgloss/libsysbase/fnmatch.c]: OK

**Comment by WinterMute** [comment]
Cool. Do you have tests for this by any chance?

**Comment by compomega** [comment]
No direct test. I just tried linking to a library that needs it. That doesn't mean I'm calling library code that actually uses it.

**Comment by WinterMute** [comment]
This might take a little while to filter through to a release. There are a couple more things currently being worked on that I want to try & get into the next release

**Pull request merged**